### PR TITLE
Support reading "development" dependencies from the installer json.

### PR DESCRIPTION
**Pull request by modmuss50**
This is needed for some upcoming loader changes.

### CODE REVIEW

File: src/main/java/net/fabricmc/loom/configuration/InstallerData.java [hunk 1]
```diff
@@ -45,35 +53,61 @@ public void applyToProject(Project project) {
...
-			project.getLogger().debug("Adding dependency ({}) from installer JSON", name);
+			LOGGER.debug("Adding dependency ({}) from installer JSON", name);
 
 			ExternalModuleDependency modDep = (ExternalModuleDependency) project.getDependencies().create(name);
-			modDep.setTransitive(false);
+			modDep.setTransitive(false); // Match the launcher in not being transitive
 			loaderDepsConfig.getDependencies().add(modDep);
 
-			// TODO: work around until https://github.com/FabricMC/Mixin/pull/60 and https://github.com/FabricMC/fabric-mixin-compile-extensions/issues/14 is fixed.
+			// Work around https://github.com/FabricMC/Mixin/pull/60 and https://github.com/FabricMC/fabric-mixin-compile-extensions/issues/14 is fixed.
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/configuration/InstallerData.java]:
```suggestion
			// Work around until https://github.com/FabricMC/Mixin/pull/60 and https://github.com/FabricMC/fabric-mixin-compile-extensions/issues/14 is fixed.
```

**Review by Juuxel — APPROVED** [review]

File: src/main/java/net/fabricmc/loom/configuration/InstallerData.java (hunk 1 above)
Comment (modmuss50) [reply src/main/java/net/fabricmc/loom/configuration/InstallerData.java]: I dont think we can ever remove this tbh, it makes sense to keep.

Comment (Juuxel) [reply src/main/java/net/fabricmc/loom/configuration/InstallerData.java]: Read the whole comment until the end 😅

Comment (modmuss50) [reply src/main/java/net/fabricmc/loom/configuration/InstallerData.java]: Good idea :D Should be better now 👍

**Pull request merged**